### ISSUE TITLE
Can't build the app after upgrading mParticle-Apple-SDK to 8.13.0

### ISSUE THREAD

**Issue by ivanszhuravlev** [mParticle/react-native-mparticle#126]
Hi everyone! Could you please help me understand what I'm doing wrong if I am?

I'm trying to upgrade the version of `mParticle-Apple-SDK` to 8.13.0, and for some reason, this issue appears:

![01_mParticle_error_small](https://user-images.githubusercontent.com/14206477/234620582-65d026e5-802f-4c74-be33-50c3cffe94a6.png)

I could've asked it in the repo of `mParticle-Apple-SDK` directly, but I'm not sure where the problem comes from, so I thought of starting here.

### What I've tried so far:
- removed `node_modules` and `Pods` folders, re-installed deps
- cleaned pods cache and DerivedData

### My Podfile:
```ruby
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '12.4'
install! 'cocoapods', :deterministic_uuids => false

is_test = ENV['MODE'] == 'e2e'
# use_flipper!()

def shared_pods
  permissions_path = '../node_modules/react-native-permissions/ios'
  pod 'Permission-AppTrackingTransparency', :path => "#{permissions_path}/AppTrackingTransparency"
  pod 'Permission-LocationWhenInUse', :path => "#{permissions_path}/LocationWhenInUse"
  pod 'Permission-Notifications', :path => "#{permissions_path}/Notifications"
  pod 'react-native-google-maps', :path => '../node_modules/react-native-maps'
  pod 'PayPalCheckout', '~> 0.88.0'
  pod 'FBSDKCoreKit'
  pod 'FBSDKLoginKit'
  pod 'TikTokOpenSDK', '~> 5.0.0'
  pod 'mParticle-Apple-SDK', '~> 8.13.0'
  pod 'mParticle-Appboy', '~> 8.1.0'

  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => true,
    :fabric_enabled => flags[:fabric_enabled],
    :flipper_configuration => FlipperConfiguration.enabled,
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  inhibit_all_warnings!
end

target 'myApp' do
  shared_pods

end

target 'myApp-staging' do
  shared_pods
end

post_install do |installer|
  react_native_post_install(
    installer,
    # Set `mac_catalyst_enabled` to `true` in order to apply patches
    # necessary for Mac Catalyst builds
    :mac_catalyst_enabled => false
  )
  __apply_Xcode_12_5_M1_post_install_workaround(installer)
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
    end
    if target.respond_to?(:product_type) and target.product_type == "com.apple.product-type.bundle"
      target.build_configurations.each do |config|
          config.build_settings['CODE_SIGNING_ALLOWED'] = 'NO'
      end
    end
  end
  installer.pods_project.build_configurations.each do |config|
    config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = "arm64"
  end
end
```

### AppDelegate

![02_imports_small](https://user-images.githubusercontent.com/14206477/234622626-169df3af-1ffe-4632-9280-588279ffe0b0.png)

<img width="725" alt="Screenshot 2023-04-26 at 17 28 20" src="https://user-images.githubusercontent.com/14206477/234625194-00cbd7e1-2e20-41ee-84c0-f2e0446f6df7.png">

### Context
- `react-native` version: 0.70.6
- `react-native-mparticle` version: 2.7.8
- initial `mParticle-Apple-SDK` version: 8.10.0

In general, I'm trying to fix the issue that appeared after I upgraded the version of RN from 0.68 to the current one and then had to bump the `react-native-mparticle` and migrate from Appboy to Braze.

I forgot to upgrade native iOS dependencies such as `mParticle-Apple-SDK` and `mParticle-Appboy` and I think that could be the reason why one specific part of the integration could be broken. In general, the data forwarding to Braze works great.

Thanks for your attention! Please let me know if I can give some more details or if I have to raise it in a different repo or even just contact the support team.

**Comment by einsteinx2** [comment]
@ivanszhuravlev @benkhlifafahmi  if you pin the mParticle-Apple-SDK to version 8.12.0 in the Podfile does that resolve the issue?

**Comment by rmi22186** [comment]
Hi all - This is a fairly old issue, and we haven't received any responses to @einsteinx2's question above from ~10 months ago.  However, a new version of `react-native-mparticle` was released - 2.7.10.  Please try using that, which fixes a few issues. We have new directions on how to implement in our [readme](https://github.com/mParticle/react-native-mparticle/blob/main/README.md) as well. 

If there are issues, please reopen this, or open a new ticket with new symptoms.  Thanks!
